src/types/item.ts

```diff
@@ -0,0 +1,34 @@
+export interface Item {
```

Comment:
What are these 3 interfaces for? And why aren't they extending one another? DRY

Reply:
They're for the surroundings getter in the store, although I don't think they'll live here. After trying it out, I've been moving away from having a top level types dir and instead having types live alongside their definitions, so inside game/items/
They should extend each other, I was just trying to find out what they actually were when defining them as they were subtly different.

Reply:
Have a non-exported ItemBase, and then export the interfaces which extend that one.
Although bear in mind that ts merges interfaces, so even without exporting them you have to be aware of naming conflicts.

Reply:
Do you have a link to any more information about that? Not quite sure what you mean by merging interfaces, and I'd like to know more.

Reply:
https://www.typescriptlang.org/docs/handbook/declaration-merging.html

Reply:
Thanks